lib/benchee/formatters/unit.ex

```diff
@@ -28,21 +35,27 @@ defmodule Benchee.Unit do
 
   @callback magnitude(unit) :: number
 
-  def float_precision(float) when float < 0.01, do: 5
-  def float_precision(float) when float < 0.1, do: 4
-  def float_precision(float) when float < 0.2, do: 3
-  def float_precision(_float), do: 2
+  @doc """
+  A separator that should appear between a value and a unit label. For example,
+  a space: `5.67 M` or an empty string: `5.67M`
+  """
+  @callback separator :: String.t
```

Comment:
You're right, it's always `" "` right now. Mostly for the reasons that you are thinking about, too.

Initally, I was imagining that counts would be formatted like in `du`: `100K`, with no space. But then I noticed that times were already formatted with a space. And I also thought about the long unit labels, and how you might want `100K` but you would probably never want `100Thousand`. So it seemed like something that

- might come down to personal preference of the user
- might need to be different for time vs count units

So, the idea is to lay the groundwork for allowing users to configure the separator as they desire (would come in #28). See [`Common.format/3`](https://github.com/PragTob/benchee/blob/95cb1900516e6d4a94a0cd2cba426ec713dc7069/lib/benchee/formatters/unit.ex#L48) (currently not exposed on the Unit modules)